### ISSUE TITLE
[Bug] admission webhook "vmariadb.kb.io" denied the request

### ISSUE THREAD

**Issue by outductor** [mariadb-operator/mariadb-operator#166]
**Describe the bug**

After upgrading the MariaDB version from 10.11.3 to 10.11.4, I keep getting the following message and the upgrade never completes.

```
admission webhook "vmariadb.kb.io" denied the request: spec.replication.primary.podIndex: Invalid value: 0: 'spec.replication.primary.podIndex' cannot be updated during a primary switchover
```

<img width="1390" alt="image" src="https://github.com/mariadb-operator/mariadb-operator/assets/137604477/98b45354-32d7-48ee-9d0c-bb9393b54452">

**Expected behaviour**

**Steps to reproduce the bug**

**Additional context**

**Environment details**:
- Kubernetes version: 1.27.4
- mariadb-operator version: 0.17.0 
- Install method: helm, OLM or static manifests
- Install flavour: minimal, recommended or custom

**Comment by mmontes11** [comment]
Hey @outductor ! This means that there is an ongoing primary switchover, so changing `spec.replication.primary.podIndex` at this point is not safe.

Could you provide the following?:
- `kubectl get mariadb <your-mariadb> -o jsonpath="{.status}"`
- Last logs of the three replicas

This is to understand why the primary swtichover hasn't finished yet. Forcefully switching the primary may imply data loss, so it is not recommended, but it can be done by disabling the validation webhook:
```bash
kubectl delete mutatingwebhookconfiguration <your-helm-release>-webhook
```

**Comment by outductor** [comment]
```
$ kubectl get mariadb -n seichi-minecraft mariadb -o jsonpath="{.status}"
{"conditions":[{"lastTransitionTime":"2023-07-20T00:41:40Z","message":"Switching primary to 'mariadb-1'","reason":"SwitchPrimary","status":"False","type":"Ready"},{"lastTransitionTime":"2023-07-15T08:36:17Z","message":"Configured replication","reason":"ConfigureReplication","status":"True","type":"ReplicationConfigured"},{"lastTransitionTime":"2023-07-20T00:41:40Z","message":"Switching primary to 'mariadb-1'","reason":"SwitchPrimary","status":"False","type":"PrimarySwitched"}],"currentPrimary":"mariadb-0","currentPrimaryPodIndex":0}
```

**Comment by mmontes11** [comment]
Alright, then this is most likely the same as this:
- https://github.com/mariadb-operator/mariadb-operator/issues/153

The previous one is about redeploying which also applies when upgrading. There is a drift between your database internal state and the `Secret` created by the operator for the `repl` user. See posible solutions in [here](https://github.com/mariadb-operator/mariadb-operator/issues/153).

**Comment by mmontes11** [comment]
This has been fixed via https://github.com/mariadb-operator/mariadb-operator/pull/178. To be released in `v0.0.20`, stay tuned!
